training_methods/automl_training_method.py

```diff
@@ -0,0 +1,440 @@
+        )
+
+        return model.resource_name
+
+    def evaluate(self, model: str) -> str:
+        """Evaluate a model and return the BigQuery table ID to its evaluation
+        table.
+
+        Args:
+            model (str): Model to evaluate.
+
+        Returns:
+            str: The BigQuery evaluation table ID.
+        """
+
+        table_id = self._evaluate(model_name=model)
+
+        return table_id
+
+    def predict(
+        self,
+        dataset: dataset.Dataset,
+        model: str,
+        model_parameters: Dict[str, Any],
+        prediction_parameters: Dict[str, Any],
+    ) -> str:
+        """Predict using a model and return the BigQuery table ID to its prediction
+        table.
+
+        Args:
+            dataset (dataset.Dataset): Input dataset.
+            model (str): Model to evaluate.
+            model_parameters (Dict[str, Any]): The model training parameters.
+            prediction_parameters (Dict[str, Any]): The prediction parameters.
+
+        Returns:
+            str: The BigQuery prediction table view ID.
+        """
+        forecast_horizon = prediction_parameters.get(FORECAST_HORIZON_PARAMETER)
+        context_window = prediction_parameters.get(CONTEXT_WINDOW_PARAMETER)
+        time_column = model_parameters.get(TIME_COLUMN_PARAMETER)
+        time_series_id_column = model_parameters.get(
+            TIME_SERIES_IDENTIFIER_COLUMN_PARAMETER
+        )
+        target_column_name = model_parameters.get(TARGET_COLUMN_PARAMETER)
+
+        if forecast_horizon is None:
+            raise ValueError(f"Missing argument: {FORECAST_HORIZON_PARAMETER}")
+
+        if context_window is None:
+            raise ValueError(f"Missing argument: {CONTEXT_WINDOW_PARAMETER}")
+
+        if time_column is None:
+            raise ValueError(f"Missing argument: {TIME_COLUMN_PARAMETER}")
+
+        if target_column_name is None:
+            raise ValueError(f"Missing argument: {TARGET_COLUMN_PARAMETER}")
+
+        # Get test data bigquery source uri
+        test_bq_source_id = dataset.get_bigquery_table_id(
+            time_column=time_column, dataset_portion="test"
+        )
+
+        processed_dataset_bq_source = self._prepare_test_dataset(
+            context_window=context_window,
+            forecast_horizon=forecast_horizon,
+            model_parameters=model_parameters,
+            bigquery_source=f"bq://{test_bq_source_id}",
+        )
+
+        prediction_table_id = self._predict(
+            model_name=model, bigquery_source=processed_dataset_bq_source
+        )
+
+        # Get the prediction dataset id
+        prediction_dataset_id = ""
+        if prediction_table_id is None:
+            raise ValueError("Prediction table id is null!")
+        else:
+            prediction_dataset_id = prediction_table_id.split(".")[1]
+
+        # Create a view and rename column names in the prediction table
+        client = bigquery.Client()
```

Comment:
I think we should be injecting this on instantiation but we can leave it for now.

Reply:
Same for the BQML training method.

Reply:
Yea while I was coding I felt that I am repeating creating a client :D